src/position/template/app-token.template.position-fetcher.ts

```diff
@@ -265,12 +265,14 @@ export abstract class AppTokenTemplatePositionFetcher<
           ]);
 
           const supply = Number(totalSupplyRaw) / 10 ** decimals;
-          if (supply === 0) return null;
 
           // Resolve price per share stage
           const pricePerShareStageFragment = { ...baseFragment, symbol, decimals, supply };
           const pricePerShareContext = { ...baseContext, appToken: pricePerShareStageFragment };
-          const pricePerShare = await this.getPricePerShare(pricePerShareContext).then(v => (isArray(v) ? v : [v]));
+          const pricePerShare =
+            supply !== 0
+              ? await this.getPricePerShare(pricePerShareContext).then(v => (isArray(v) ? v : [v]))
+              : Array(baseFragment.tokens.length).fill(1);
```

Comment:
Not a safe change. Instead:
1. Revert https://github.com/Zapper-fi/studio/pull/2770/files
2. For the app that had NaN price per share (I can’t remember which one), add the same “supply === 0” check as seen in that reverted PR.